### PR TITLE
LM-178 Fixed token caching mechanism to use home directory

### PR DESCRIPTION
**Pull request by dusktreader**
#### What
Adjusted token caching code to use the current user's home directory.

#### Why
The agent was having issues with root and the slurm user both trying to use the same token.

`Task`: https://app.clickup.com/t/18022949/LM-178

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

File: agent/lm_agent/backend_utils.py [hunk 1]
```diff
@@ -12,6 +13,20 @@
 from lm_agent.logs import logger
 
 
+def _get_cache_dir() -> pathlib.Path:
+    """
+    Get the path to the cache dir. Will always be a sub-directory of the current user's home.
+    """
+    return pathlib.Path.home() / f".cache/license-manager"
```
Comment (heitorPB) [inline agent/lm_agent/backend_utils.py]: LM user is a system user, not a normal one. `$HOME` might  not be defined and we might not have permissions to create this folder, specially in a shared environment with network mounted `/home`.

Why changing from `/run`?

Comment (dusktreader) [reply agent/lm_agent/backend_utils.py]: Addressed this issue in the latest commit. CACHE_DIR restored as an environment variable. No logic to create missing directories. If the cache dir doesn't exist, you don't get cached tokens.

File: agent/lm_agent/config.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 
 DEFAULT_DOTENV_PATH = Path("/etc/default/license-manager-agent")
+DEFAULT_CACHE_DIR = Path.home() / Path(".cache/license-manager")
```
Comment (heitorPB) [inline agent/lm_agent/config.py:14]:
```suggestion
DEFAULT_CACHE_DIR = Path("/var/cache/license-manager")
```
?

Comment (dusktreader) [reply agent/lm_agent/config.py:14]: No, it needs to default to home for running locally. In the charm, it is set to `/var/cache/license-manager`

Comment (heitorPB) [reply agent/lm_agent/config.py:14]: It took me some time to understand how the code does that. Could you simplify it to make it explicit? There's no need to have two variables to hold the cache directories. This is a huge source of confusion right now.

Comment (dusktreader) [reply agent/lm_agent/config.py:14]: I don't quite understand. The `DEFAULT_CACHE_DIR` is just the default setting. The final value will either be the same or can be overriden by setting the environment.

**Review by julianaklulo — APPROVED** [review]

**Pull request merged**